File: src/main/webapp/app/order/order.service.js
(function () {
	'use strict';

	angular
		.module('main')
		.service('OrderService', function ($http) {

			var urlBase = '/order';

			this.getByUsername = function (username) {
				return $http.get(urlBase + '/' + username);
			};

			this.getCost = function(username) {
				return $http.get(urlBase + '/cost/' + username);
			};

			this.removeGame = function(id) {
				return $http.post(urlBase + '/game/delete/' + id);
			};

		});
		
})();

File: src/main/java/computer/games/order/service/OrderService.java
package computer.games.order.service;

import computer.games.game.domain.Game;
import computer.games.order.domain.Order;
import org.springframework.http.ResponseEntity;

import java.util.List;

/**
 * Created by Ivan on 26.01.2017.
 */
public interface OrderService {

    ResponseEntity<Order> save(String gameName, String username);

    ResponseEntity<List<Game>> getByUsername(String username);

    ResponseEntity<Double> getCost(String username);

    ResponseEntity<Void> deleteGameFromOrder(Long id);

}
